operator/v1/types_ingress.go

```diff
@@ -925,6 +934,26 @@ type IngressControllerHTTPHeaders struct {
 	HeaderNameCaseAdjustments []IngressControllerHTTPHeaderNameCaseAdjustment `json:"headerNameCaseAdjustments,omitempty"`
 }
 
+// IngressControllerHTTPHeaderBuffer specifies the size of the
+// per-connection HTTP header buffers.
+type IngressControllerHTTPHeaderBuffer struct {
+	// headerBufferSize describes how much memory should be reserved
+	// (in bytes) for IngressController connection sessions.
+	//
+	// +kubebuilder:validation:required
+	// +required
```

Comment:
In https://github.com/openshift/enhancements/pull/449 @Miciah and I agreed that it would make sense to validate these values before applying them in the ingress operator so cluster admins cant brick the default ingress controller. Unclear to me if we should set some absolute bare minimum integer value here as well (or what those values would be).

Reply:
But you won't change it from the original defaults, anyway, because that could cause problems in previously stable configurations.

Reply:
> But you won't change it from the original defaults, anyway, because that could cause problems in previously stable configurations.

Right, if a user does not set `HTTPHeaderBuffer`, then the original default values in the router template will be used.